### ISSUE TITLE
SpotBugs Should Detect Infinite Loop with Boolean.TRUE

### ISSUE THREAD

**Issue by RJerrica** [spotbugs/spotbugs#3758]
Hi, I found a false negative related to the `IL_INFINITE_LOOP` rule in the following code example. SpotBugs should report a warning at line 3, as this is clearly an infinite loop. Additionally, `Boolean.TRUE` is widely used in real-world projects, so SpotBugs should detect this case.

### Minimized Code Example
```java
public class Main {
    public static void main(String[] args) {
        while (Boolean.TRUE) {
            System.out.println("Wow");
        }
    }
}
```

SpotBugs version: 4.9.6

**Comment by gtoison** [comment]
Hello, this is the same issue as #3460 and there was a short discussion here #3537 
Essentially these unconditional infinite loops are not detected as bugs on purpose, because this would raise too many false positives
